src/set_theory/fixed_points.lean

```diff
@@ -0,0 +1,316 @@
+/-
+Copyright (c) 2022 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+
+import set_theory.ordinal_arithmetic
+
+/-!
+# Fixed points of normal functions
+
+We prove various statements about the fixed points of normal ordinal functions. We state them in
+three forms: as statements about type-indexed families of normal functions, as statements about
+ordinal-indexed families of normal functions, and as statements about a single normal function. For
+the most part, the first case encompasses the others.
+
+Moreover, we prove some lemmas about the fixed points of specific normal functions.
+
+## Main definitions and results
+
+* `nfp_family`, `nfp_bfamily`, `nfp`: the next fixed point of a (family of) normal function(s).
+* `fp_family_unbounded`, `fp_bfamily_unbounded`, `fp_unbounded`: the (common) fixed points of a
+  (family of) normal function(s) are unbounded in the ordinals.
+* `deriv_add_eq_mul_omega_add`: a characterization of the derivative of addition.
+-/
+
+noncomputable theory
+
+universes u v
+
+open function
+
+namespace ordinal
+
+/-! ### Fixed points of type-indexed families of ordinals -/
+
+section
+variables {ι : Type u} {f : ι → ordinal.{max u v} → ordinal.{max u v}}
+
+/-- The next common fixed point, at least `a`, for a family of normal functions. -/
```

Comment:
I think it would be nice in this docstring to name the lemmas which give the vital properties of this function: namely that this is a fixed point of each f, that it's at least `a`, and that it's the smallest thing such that those two are true